docker/Dockerfile

```diff
@@ -1,7 +1,24 @@
 ARG PYTHON_VERSION=2
-FROM docker.io/python:${PYTHON_VERSION}-alpine3.8
 
-COPY . /synapse
+###
+### Stage 0: builder
+###
+FROM docker.io/python:${PYTHON_VERSION}-alpine3.8 as builder
+
+# first of all, install the runtime deps; this layer can then be used in the
+# final stage as well.
+
+RUN apk add --no-cache --virtual .runtime_deps \
+        libffi \
+        libjpeg-turbo \
+        libressl \
+        libxslt \
+        libpq \
+        zlib \
+        su-exec
```

Comment:
I'm relatively sure we can skip over this whole section because all of those are dependencies of the build dependencies anyway. Listing them explicitly was only required because we removed the build dependencies, which also removed those if they were not explicitly installed separately.

Considering we're not uninstalling any packages but throwing away containers instead, we can even skip the whole `--virtual .*_deps` part, because that is only needed for easier deletion of the packages.